src/lib/crypto/kimchi_bindings/stubs/binding_generation/src/main.rs

```diff
@@ -35,143 +36,176 @@ use wires_15_stubs::{
 fn main() {
     let args: Vec<String> = std::env::args().collect();
 
-    if let Some(output_file) = args.get(1) {
-        let mut file = File::create(output_file).expect("could not create output file");
-        generate_bindings(&mut file);
+    let env = &mut Env::default();
+
+    let header = "(* This file is generated automatically with ocaml_gen. *)\n";
+
+    if let Some(kimchi_types) = args.get(1) {
+        let mut file = File::create(kimchi_types).expect("could not create output file");
+        write!(file, "{}", header).unwrap();
+        let _ = env.new_module("Kimchi_types");
+        generate_types_bindings(&mut file, env);
+        let _ = env.parent();
     } else {
         let mut w = std::io::stdout();
-        generate_bindings(&mut w);
+        write!(w, "{}", header).unwrap();
+        decl_module!(w, env, "Kimchi_types", {
+            generate_types_bindings(&mut w, env);
+        });
     };
+    if let Some(pasta_bindings) = args.get(2) {
+        let mut file = File::create(pasta_bindings).expect("could not create output file");
+        write!(file, "{}", header).unwrap();
+        let _ = env.new_module("Pasta_bindings");
+        generate_pasta_bindings(&mut file, env);
+        let _ = env.parent();
```

Comment:
Interesting, I was wondering how the bindings from Kimchi_bindings were going to refer to the bindings from Pasta_bindings. So I guess you fake the module here, I'm happy it works without any changes to ocaml-gen :)